### PR TITLE
PanelLibrary: Adds library panel meta information to dashboard json

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR adds a counter of connected dashboards to Library Panel meta information so that the frontend doesn't need to do additional calls to the backend / panel.
This PR also adds meta information to library panels during loading of a dashboard.
This PR also adds more test cases and checks the responses in more detail.

**Which issue(s) this PR fixes**:

Relates #29610

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/librarypanels/database.go
Comment (hugohaggmark) [inline pkg/services/librarypanels/database.go:23]: Couldn't figure out how to write a better SQL statement here. Couldn't get the Join to work when there are no connections for a Library Panel and dashboard.

File: pkg/services/librarypanels/librarypanels_test.go
Comment (hugohaggmark) [inline pkg/services/librarypanels/librarypanels_test.go:71]: I'm thinking about extracting this out to it's own function. Is this something that I should do or is it more Go to keep it like this?

**Review by torkelo — APPROVED** [review]
Looks good!

**Comment by hugohaggmark** [comment]
Waiting for some feedback from @papagian and @kylebrandt before merging this.

**Pull request merged**
